File: app/src/main/java/com/pampang/nav/fcm/FCMAuthInterceptor.java
package com.pampang.nav.fcm;

import android.content.Context;
import androidx.annotation.NonNull;
import java.io.IOException;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class FCMAuthInterceptor implements Interceptor {
    private final Context context;

    public FCMAuthInterceptor(Context context) {
        this.context = context.getApplicationContext();
    }

    @NonNull
    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {
        String token = FCMTokenHelper.getAccessToken(context);

        if (token == null) {
            throw new IOException("Failed to get Firebase access token");
        }

        Request originalRequest = chain.request();
        Request newRequest = originalRequest.newBuilder()
                .addHeader("Authorization", "Bearer " + token)
                .addHeader("Content-Type", "application/json; UTF-8")
                .build();

        return chain.proceed(newRequest);
    }
}

File: app/src/main/java/com/pampang/nav/network/FCMAuthInterceptor.kt
package com.pampang.nav.network

import android.content.Context
import com.pampang.nav.utils.FCMTokenHelper
import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response
import java.io.IOException

class FCMAuthInterceptor(private val context: Context) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val token = FCMTokenHelper.getAccessToken(context)
                ?: throw IOException("Failed to get Firebase access token")

        val originalRequest: Request = chain.request()
        val newRequest: Request = originalRequest.newBuilder()
                .addHeader("Authorization", "Bearer $token")
                .addHeader("Content-Type", "application/json; UTF-8")
                .build()

        return chain.proceed(newRequest)
    }
}
